build-tools-internal/src/main/groovy/elasticsearch.build-complete.gradle

```diff
@@ -64,5 +65,27 @@ if (buildNumber && performanceTest == null && GradleUtils.isIncludedBuild(projec
     } catch (Exception e) {
       logger.lifecycle("Failed to archive additional logs", e)
     }
+
+    if (uploadFile.exists() && System.getenv("BUILDKITE") == "true") {
+      try {
+        println "Uploading buildkite artifact: ${uploadFilePath}..."
+        new ProcessBuilder("buildkite-agent", "artifact", "upload", uploadFilePath)
+          .start()
+          .waitFor()
+
+        println "Generating buildscan link for artifact..."
+
+        def process = new ProcessBuilder("buildkite-agent", "artifact", "search", uploadFilePath, "--step", System.getenv('BUILDKITE_JOB_ID'), "--format", "%i").start()
```

Comment:
It's very annoying that `artifact upload` doesn't just return the id and we have to do a separate query here.